pkg/monitor/datapath_trace.go

```diff
@@ -41,10 +43,11 @@ const (
 const (
 	TraceNotifyVersion0 = iota
 	TraceNotifyVersion1
+	TraceNotifyVersion2
 )
 
-// TraceNotify is the message format of a trace notification in the BPF ring buffer
-type TraceNotify struct {
+// TraceNotifyV0 is the common message format for versions 0 and 1.
+type TraceNotifyV1 struct {
```

Comment:
Nit: This struct seems to only ever be used embedded in `TraceNotifyV2`. 

Maybe I'm missing something, but you could have just added a new field and the `Decode` logic would still  work fine (which seems to be the way we handled V0->V1)